### DISCUSSION TITLE
PaddleX/latest/module_usage/tutorials/ocr_modules/doc_img_orientation_classification

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleX#2603] [General]
# PaddleX/latest/module_usage/tutorials/ocr_modules/doc_img_orientation_classification

https://paddlepaddle.github.io/PaddleX/latest/module_usage/tutorials/ocr_modules/doc_img_orientation_classification.html?h=%E5%88%86

**Comment by jiauy** [comment]
效果不如paddlecls的文本方向分类模型

**Reply by jiauy** [reply]
说错了，是图像方向分类模型，用的不是同一个模型吧

**Reply by cuicheng01** [reply]
用的是同一个模型哈

**Reply by SWHL** [reply]
正想测试一下和paddlecls哪个效果好呢。原来是一个，晓得了。

**Comment by beer168** [comment]
官方例子跑出来的旋转角度都不对

**Reply by beer168** [reply]
建议用自己数据集重新训练

**Reply by cuicheng01** [reply]
当时可能有误传权重，很快下掉了，建议清理cache中的权重，重新体验

**Comment by shuaimoumou** [comment]
这模型准确度也太差了吧？4张图有3张都是错的。。。

**Reply by cuicheng01** [reply]
当时可能有误传权重，很快下掉了，建议清理cache中的权重，重新体验

**Comment by loveyu** [comment]
默认模型准确度太差，试了几个都是错的

**Reply by cuicheng01** [reply]
当时可能有误传权重，很快下掉了，建议清理cache中的权重，重新体验

**Reply by yuanmengwei** [reply]
还是不行呀，方向分类的精度还是有问题

**Comment by ty1ooeujay** [comment]
'PP-LCNet_x1_0_doc_ori' is not a registered model name.这个怎么处理

**Reply by cuicheng01** [reply]
可以参考PaddleX中的安装文档，如安装PaddleClas插件，`paddlex --install PaddleClas`

**Comment by beer168** [comment]
前一步没安装，使用前还要安装paddle对应的类型插件，如PaddleClas,Seg之类的



---Original---
From: ***@***.***&gt;
Date: Wed, Feb 26, 2025 11:02 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [PaddlePaddle/PaddleX]PaddleX/latest/module_usage/tutorials/ocr_modules/doc_img_orientation_classification(Discussion #2603)




 
'PP-LCNet_x1_0_doc_ori' is not a registered model name.这个怎么处理

**Comment by cddddd6** [comment]
import base64
from sanic import Sanic, response
from paddleocr import PaddleOCR
import re
import numpy as np
import cv2
from collections import Counter
app = Sanic(__name__)

# 模型目录
det_model_dir = 'E:/users/chend/PycharmProjects/pythonProject1/inference_model/detv4_ess_cv_bmu/'
rec_model_dir = 'E:/users/chend/PycharmProjects/pythonProject1/inference_model/openatom_rec_svtrv2_ch_infer/'
cls_model_dir = 'E:/users/chend/PycharmProjects/pythonProject1/inference_model/clsv2/'

# 初始化OCR引擎
ocr = PaddleOCR(
    det_model_dir=det_model_dir,
    rec_model_dir=rec_model_dir,
    cls_model_dir=cls_model_dir,
    use_angle_cls=True,
    det_db_unclip_ratio=0.4
)

请教下能否用上面的代码调用PP-LCNet_x1_0_doc_ori模型

**Reply by cddddd6** [reply]
我在运行以下代码时报错，自动Connecting to https://paddle-model-ecology.bj.bcebos.com/paddlex/PaddleX3.0/fonts/PingFang-SC-Regular.ttf，但我的环境是离线的，无法连接，请问能否指定字体的路径，谢谢！

from paddlex import create_model

model = create_model(model_name="PP-LCNet_x1_0_doc_ori",
                     model_dir="E:/users/chend/PycharmProjects/pythonProject1/inference_model/PP-LCNet_x1_0_doc_ori_infer",
                     save_dir=None)
output = model.predict("E:/users/chend/Desktop/bmu_wrong",  batch_size=1)
for res in output:
    res.print(json_format=False)
    res.save_to_img("E:/users/chend/Desktop/bmu_wrong/demo.png")
    res.save_to_json("E:/users/chend/Desktop/bmu_wrong/res.json")

**Comment by nainai123** [comment]
我根据二次开发流程在linux上对模型进行训练，在图片校验没问题 ，但是在执行训练命令的时候收到了报错提示
/home/yuzai/.conda/envs/paddlex/bin/python: can't open file '/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/repo_manager/repos/PaddleClas/tools/train.py': [Errno 2] No such file or directory
Traceback (most recent call last):
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/utils/result_saver.py", line 29, in wrap
    result = func(self, *args, **kwargs)
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/engine.py", line 41, in run
    self._model.train()
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/model.py", line 101, in train
    trainer.train()
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/modules/base/trainer.py", line 88, in train
    train_result = self.pdx_model.train(**train_args)
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/repo_apis/PaddleClas_api/cls/model.py", line 126, in train
    return self.runner.train(
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/repo_apis/PaddleClas_api/cls/runner.py", line 50, in train
    return self.run_cmd(
  File "/home/yuzai/imageClass/PaddleX-3.0.0-rc0/paddlex/repo_apis/base/runner.py", line 356, in run_cmd
    raise CalledProcessError(
paddlex.utils.errors.others.CalledProcessError: Command ['/home/yuzai/.conda/envs/paddlex/bin/python', 'tools/train.py', '-c', '/home/yuzai/.paddlex/tmpt78jh7dd/clsmodel_PP-LCNet_x1_0_doc_ori.yml', '-o', 'Global.eval_during_train=True'] returned non-zero exit status 2.
看起来时缺少PaddleClas文件夹以及以下的一些文件 怎么解决这个问题

**Reply by nainai123** [reply]
这个错时没安装插件导致的   没想到时这么安装的插件paddlex --install PaddleOCR PaddleClas

**Comment by FSA9** [comment]
好像对于表格和少文本行的文档识别效果不太行啊，有什么方法补救吗？+size？
以及为什么显卡占用率这么低？batchsize开了2000结果利用率基本不到百分之十，也没有CPU和IO瓶颈

**Comment by dhhcj1** [comment]
都2025年了，做四个方向的分类能做成这样。。。我怀疑自己了好几个小时

**Comment by uilstong** [comment]
这个支持用paddlex --server使用吗？我想通过URL使用

**Comment by YoungJulY0728** [comment]
请问以下各位大佬，就是在4.1.2 数据校验中：
python main.py -c paddlex/configs/modules/doc_text_orientation/PP-LCNet_x1_0_doc_ori.yaml \
    -o Global.mode=check_dataset \
    -o Global.dataset_dir=./dataset/text_image_orientation
这个main.py是什么文件啊，在哪里呢，我没有找到               ：（

**Comment by yuancc123456** [comment]
paddlex.utils.errors.others.UnsupportedParamError: 'PP-LCNet_x1_0_doc_ori' is not a registered model name.好几个模块都这样的报错
